src/components/form-field/search-field/search-field.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useRef } from "react";
+import { classNames } from "../../../util/class-names";
+import CrossIcon from "../../../icons/cross-icon";
+import SearchIcon from "../../../icons/search-icon";
+
+export interface SearchInputProps extends React.ComponentPropsWithoutRef<"input"> {
+    autoSelect?: boolean;
+    ariaDescribedBy?: string;
+    error?: boolean;
+    onClear: () => void;
+}
+
+export const SearchInput = ({
+    ariaDescribedBy,
+    readOnly,
+    autoSelect,
+    onClear,
+    error,
+    value,
+    disabled,
+    className,
+    ...props
+}: SearchInputProps) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const isClearIconShown = useMemo(() => {
+        return !readOnly && !disabled && value?.toString().length;
+    }, [disabled, readOnly, value]);
+
+    const handleAutoSelection = () => {
+        if (autoSelect && inputRef.current) {
+            inputRef.current.select();
+        }
+    };
+
+    return (
+        <div className={classNames("relative w-full")}>
+            <div
+                className="pointer-events-none absolute inset-y-0 left-0 z-10 flex items-center pl-3"
+                aria-hidden="true"
+            >
+                <SearchIcon className="text-gray-400 h-3.5 w-3.5 fill-neutral-600" />
+            </div>
+
+            <input
+                ref={inputRef}
+                aria-describedby={ariaDescribedBy}
+                onMouseOver={handleAutoSelection}
+                onFocus={handleAutoSelection}
+                onClick={handleAutoSelection}
+                type="search"
```

Comment:
In some browsers this type creates a default reset button:
![CleanShot 2023-11-07 at 09 47 55@2x](https://github.com/abusix/hailstorm/assets/3883653/096916c0-225d-4a8d-ae2f-0a9601779489)

Reply:
thank you for that example tailwind link!